core/src/test/java/org/elasticsearch/ingest/core/CompoundProcessorTests.java

```diff
@@ -108,4 +114,30 @@ public void testSingleProcessorWithNestedFailures() throws Exception {
         assertThat(processorToFail.getInvokedCounter(), equalTo(1));
         assertThat(lastProcessor.getInvokedCounter(), equalTo(1));
     }
+
+    public void testSetSplitAppend() throws Exception {
+        TemplateService templateService = TestTemplateService.instance();
```

Comment:
wasn't sure whether we should create other processors within the `SplitProcessorTests`. then yes, this can be moved there